src/Workspaces/Core/Portable/Utilities/EditDistance.cs

```diff
@@ -1,175 +1,507 @@
 
-                // The edit distance is the last element in the current row
-                return rowCurrent[rowWidth - 1];
-            }
-            finally
+            // If our threshold is greater than the number of edits it would take to just produce 'longString'
+            // then just cap the threshold.  In the code below 'threshold' acts as the diagonal
+            // we don't have to look outside of.  By capping the threshold here we make the math
+            // easy below.
+            if (costThreshold > longLength)
             {
-                ReleaseArray(rowBefore2);
-                ReleaseArray(rowBefore1);
-                ReleaseArray(rowCurrent);
+                costThreshold = longLength;
             }
-        }
 
-        // =============================================================================
-        // Get the length of the longest common subsequence of the two given strings
-        // =============================================================================
-        public static int GetLongestCommonSubsequenceLength(string oldString, string newString)
-        {
-            // Expand the counts by one to include the empty string
-            int rowWidth = oldString.Length + 1;
-            int columnHeight = newString.Length + 1;
+            // If the number of edits we'd have to perform is greater than our threshold, then
+            // there's no point in even continuing.
+            if (minimumEditCount > costThreshold)
+            {
+                return int.MaxValue;
+            }
 
-            // This stores the row before the current one
-            int[] rowBefore1 = GetArray(rowWidth);
+            //  The matrix we are virtually simulating here is indexed in the following fashion:
+            //
+            //      i<- 0 1 2 3 4 5 6 7
+            //    j     X l m R e a d e
+            //    ^  
+            //    0 X
+            //    1 m
+            //    2 l
+            //    3 R
+            //    4 e
+            //    5 a
+            //    6 d
+            //    7 e
+            //    8 r
+            //
+            //          It will be virtually initialized with the following values.  These represent the
+            //          initial 'left' and 'above' values for the algorithm below
+            //
+            //      i<-|0 1 2 3 4 5 6 7
+            //    j    |X l m R e a d e
+            //    ^   0|1 2 3 4 5 6 7 8
+            //    -----+---------------
+            //    0 X 1|
+            //    1 m 2|
+            //    2 l 3|
+            //    3 R 4|
+            //    4 e 5|
+            //    5 a 6|
+            //    6 d 7|
+            //    7 e 8|
+            //    8 r 9|
+            //
+            //          We'll then proceed a column at a time from left ot right producing the new column values.
+            //          In the absense of twiddles we can note something useful:  The value for any given (i,j) we
```

Comment:
Is "twiddle" a generally understood term in the literature, or should we have a definition.